backend/cmd/headlamp.go

```diff
@@ -275,6 +276,30 @@ func createHeadlampHandler(config *HeadlampConfig) http.Handler {
 
 	config.handleClusterRequests(r)
 
+	r.HandleFunc("/externalapiproxy", func(w http.ResponseWriter, r *http.Request) {
+		url, err := url.Parse(r.Header.Get("ExternalAPI"))
+		if err != nil {
+			log.Fatal("Failed to get URL from server", err)
+		}
+		isURLContainedInProxyURLs := false
+		for _, proxyURL := range config.proxyURLs {
+			if strings.HasPrefix(url.Host, proxyURL) {
+				isURLContainedInProxyURLs = true
+			}
+		}
+		if !isURLContainedInProxyURLs {
+			http.Error(w, "no allowed proxy url match, request denied ", http.StatusBadRequest)
+		}
```

Comment:
From our previous discussion, I supposed this will be a wildcard match instead of a Host match.

FYI, this is how we check for valid redirect URLs in Nebraska.

https://github.com/kinvolk/nebraska/blob/main/backend/pkg/auth/oidc.go#L248-L258